src/main/java/game/Submarine.java

```diff
@@ -149,28 +151,52 @@ public void executeStrategy() {
 			 * */
 
 				if (!map.enemyShips.isEmpty()) {
-
-					// TODO: a legmesszebbit kene ?
-					Vector2D enemyShip = new Vector2D(map.enemyShips.get(0).x(), map.enemyShips.get(0).y());
-
-					//current vector
-					Vector2D currentVector = new Vector2D(this.position.getX(), this.position.getY());
-
-					//target Vector
-					Vector2D targetVector2 = enemyShip.subtract(currentVector);
-
-					double targetVectorAngle2 = getAngleOfVector(targetVector2);
-					if (targetVectorAngle2 < 0) {
-						targetVectorAngle2 += 360;
-					}
-					log.info("Ship[{}]: TargetVectorAngle calculated: {}", this.id, targetVectorAngle2);
-
-					if (this.torpedoCooldown <= 0) {
-						this.actionQueue.add(Action.shoot(targetVectorAngle2));
-					}
+				
+				// TODO: a legmesszebbit kene ?
+
+				ProjectileLike enemy = map.enemyShips.get(0);
+				Vector2D enemyShip = new Vector2D(enemy.x(), enemy.y());
+				
+				//current vector
+				Vector2D currentVector = new Vector2D(this.position.getX(), this.position.getY());
+				
+				//target Vector
+				Vector2D targetVector2 = enemyShip.subtract(currentVector);
+
+				
+				double distanceFromTarget = currentVector.distance(targetVector2);
+				int numberOfRoundsToReachTarget = (int) Math.ceil(distanceFromTarget/this.map.mapConfig.torpedoSpeed);
```

Comment:
This is saved to the submarine class as TORPEDO_SPEED. Check out the class-static variables at the top.